code/game/machinery/atmoalter/clamp.dm

```diff
@@ -0,0 +1,150 @@
+//Good luck. --BlueNexus
+
+//Static version of the clamp
+/obj/machinery/clamp
+	name = "stasis clamp"
+	desc = "A magnetic clamp which can halt the flow of gas in a pipe, via a localised stasis field."
+	icon = 'icons/atmos/clamp.dmi'
+	icon_state = "pclamp0"
+	anchored = 1.0
+	var/obj/machinery/atmospherics/pipe/simple/target = null
+	var/open = 1
+
+	var/datum/pipe_network/network_node1
+	var/datum/pipe_network/network_node2
+
+/obj/machinery/clamp/New(loc, var/obj/machinery/atmospherics/pipe/simple/to_attach = null)
+	..()
+	if(istype(to_attach))
+		target = to_attach
```

Comment:
Multiple clamps attached to the same target will interfere with each other and potentially act unpredictably